packages/web-components/src/components/ic-search-bar/ic-search-bar.tsx

```diff
@@ -257,85 +265,42 @@ export class SearchBar {
-  async setFocus(): Promise<void> {
-    this.retryViaKeyPress = false;
-    this.retryButtonClick = false;
-    if (this.inputEl) {
-      this.inputEl.setFocus();
-    }
-  }
-
-  @Listen("icKeydown", {})
-  handleKeyDown(ev: CustomEvent): void {
-    const keyEv: KeyboardEvent = ev.detail.event;
-    if (this.menu && this.open) {
-      this.menu.handleKeyboardOpen(keyEv);
-    }
-  }
-
-  @Listen("keyup", {})
-  handleKeyUp(ev: KeyboardEvent): void {
-    if (ev.key === "Enter") {
-      if (this.preventSubmit || this.isSubmitDisabled()) {
-        return;
-      }
-
-      this.handleSubmitSearch();
-      this.setMenuChange(false);
-    }
-
-    if (ev.key === "Escape") {
-      this.setMenuChange(false);
-    }
-
-    if (this.preventSubmit) {
-      this.preventSubmit = false;
-    }
-  }
-
-  private handleSubmitSearch = () => {
-    this.highlightedValue && (this.value = this.highlightedValue);
-    this.highlightedValue = undefined;
-    this.icSubmitSearch.emit({ value: this.value });
+  @Event() icChange: EventEmitter<IcValueEventDetail>;
 
-    const form: HTMLFormElement = this.el.closest("FORM");
+  /**
+   * Emitted when value is cleared with clear button
+   */
+  @Event() icClear: EventEmitter<void>;
+  private handleClear = (ev: Event) => {
+    const keyboardEvent = ev as KeyboardEvent;
+    const mouseEvent = ev as MouseEvent;
 
-    if (this.searchSubmitButton && !!form && !this.preventSubmit) {
-      handleHiddenFormButtonClick(form, this.searchSubmitButton);
-    }
-  };
+    if (
+      mouseEvent.type === "click" ||
+      keyboardEvent.code === "Enter" ||
+      keyboardEvent.code === "Space"
+    ) {
+      this.value = "";
+      this.inputEl.value = "";
+      this.loading = false;
+      clearTimeout(this.timeoutTimer);
+      this.filteredOptions = this.options;
+      this.el.setFocus();
 
-  private handleSubmitSearchKeyDown = (ev: KeyboardEvent) => {
-    if (ev.key === " ") {
+      this.icClear.emit();
       ev.preventDefault();
-      this.handleSubmitSearch();
-    }
-  };
 
-  private handleRetry = (ev: CustomEvent<IcValueEventDetail>) => {
-    this.retryViaKeyPress = ev.detail.keyPressed === "Enter";
-    this.icRetryLoad.emit({ value: ev.detail.value });
-    this.triggerLoading();
-    this.retryButtonClick = true;
+      this.preventSubmit = true;
+    }
   };
 
-  private triggerLoading = () => {
-    const loadingOption: IcMenuOption[] = [
-      { label: this.loadingLabel, value: "", loading: true },
-    ];
-    if (this.filteredOptions !== loadingOption)
-      this.filteredOptions = loadingOption;
-    if (this.timeout) {
-      this.timeoutTimer = window.setTimeout(() => {
-        this.filteredOptions = [
-          { label: this.loadingErrorLabel, value: "", timedOut: true },
-        ];
-      }, this.timeout);
-    }
+  private handleMouseDown = (ev: Event) => {
+    ev.preventDefault();
   };
 
   // The icInput event is defined here so that it appears as an event for search bar
```

Comment:
debounceAriaLiveUpdate (line 340) needs moving to be with the other local private methods